schema/crdb/dbinit.sql

```diff
@@ -1339,9 +1339,15 @@ CREATE TABLE IF NOT EXISTS omicron.public.oximeter (
     time_created TIMESTAMPTZ NOT NULL,
     time_modified TIMESTAMPTZ NOT NULL,
     ip INET NOT NULL,
-    port INT4 CHECK (port BETWEEN 0 AND 65535) NOT NULL
+    port INT4 CHECK (port BETWEEN 0 AND 65535) NOT NULL,
+    time_deleted TIMESTAMPTZ
 );
 
+CREATE UNIQUE INDEX IF NOT EXISTS lookup_non_deleted_oximeter ON omicron.public.oximeter (
+    id
+) WHERE
+    time_deleted IS NULL;
+
```

Comment:
I was trying to figure out why it's here and:

- the id is already a primary key so it's not needed to ensure id uniqueness among non-deleted rows
- it doesn't help you look up any single row because if you have the id, the primary key index lets the database go straight to it and it can simply not return it if `time_deleted IS NULL`
- but it would help if the goal was to list non-deleted instances ordered by id

If it's just for that last case then a different name and maybe a comment would help.

Reply:
Yeah, the blunt reason for why it's here is "I got a no table scans allowed error running existing tests". I think it is the final bullet (listing non-deleted instances). How about `list_non_deleted_oximeter` plus a comment?